src/main/java/com/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.UserDao;
+import com.cinema.exceptions.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.User;
+import com.cinema.util.HibernateUtil;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
```

Comment:
```suggestion
        try {
            Session session = HibernateUtil.getSessionFactory().openSession();
             Transaction transaction = session.beginTransaction();
            session.save(user);
```
I didn't mean try-with recourses)

Reply:
Yeah, I understand - but I close session in finally block, and roll back a transaction in catch block. I can't do that if I move those inside the try block - so may I decline this suggestion?)